pkg/utils/utils.go

```diff
@@ -318,22 +336,32 @@ func MonitorMounts() {
 			}
 			TransitionVolListLock.Unlock()
 		}
+		if cleanupRequired {
+			Cleanup()
+		}
 	}
 }
 
-// CleanupOnRestart unmounts and detaches the volumes having
+// Cleanup unmounts and detaches the volumes having
 // DeletionTimestamp set and removes finalizers from the
 // corresponding CStorVolumeAttachment CRs
-func CleanupOnRestart() {
+func Cleanup() (err error) {
 	var (
-		err        error
 		csivolList *apis.CStorVolumeAttachmentList
 	)
 	// Get list of mounted paths present with the node
 	TransitionVolListLock.Lock()
 	defer TransitionVolListLock.Unlock()
-	if csivolList, err = GetVolListForNode(); err != nil {
-		return
+	count := 0
+	for {
+		if csivolList, err = GetVolListForNode(); err == nil {
```

Comment:
If this error out more than 5 times then the list will be empty and the error will not be logged as well. Can we log the error outside the for loop in that case.

Reply:
If it errors 5 times, we are returning an error to the caller function.